### PR TITLE
feature(component): Change `Accordion`s to close inactive panels, resolves #159

### LINKED ISSUE #159: `Accordion`s should only keep active item open by default
**Describe the bug**
By default, `Accordion`s should close any `Accordion.Panel`s that are not the most recently clicked (*active*). See the Flowbite Accordion docs.

**To Reproduce**
1. View an Accordion
2. Click any other element than the initially active element
3. Observe that both items are still open (the initially active should have closed now)

**Expected behavior**
By default, `Accordion`s should close any panels that are not the most recently clicked.

This should be a prop... `multi` to enable keeping more than one? Ideas on that? @rluders @bacali95

### PR DESCRIPTION
**Pull request by tulup-conner**
## Breaking changes

`Accordion`s used to let more than one `Accordion.Panel` be open at once. Now, only the most recently clicked panel will be open.

To override this behavior, specify `<Accordion alwaysOpen>`.

See the [Flowbite Accordion docs](https://flowbite.com/docs/components/accordion/) - notably, the `Default accordion` and `Always open` examples

## Features

- [x] [feature(component): Change Accordions to close inactive panels, res…](https://github.com/themesberg/flowbite-react/commit/58ab0f1b13732d691ee63542343d4bad8766625d)
- [x] [feature(story): Add missing Accordion stories](https://github.com/themesberg/flowbite-react/commit/e6d7e2d1927e98bedf233c15384ed3f9e0e1d4de)

### Unit

- [x] [test(component): Add numerous unit tests for Accordions](https://github.com/themesberg/flowbite-react/commit/7e54b587539cd438b9ce9c99e56cd3e96201d14f)

### CODE REVIEW

File: src/lib/components/Accordion/AccordionContent.tsx [hunk 1]
```diff
@@ -7,11 +7,11 @@ import { useAccordionContext } from './AccordionPanelContext';
 export const AccordionContent: FC<ComponentProps<'div'>> = ({ children, ...props }): JSX.Element => {
   const theirProps = excludeClassName(props);
 
-  const { isOpen } = useAccordionContext();
+  const { open } = useAccordionContext();
```
Comment (rluders) [inline src/lib/components/Accordion/AccordionContent.tsx:10]: isOpen

File: src/lib/components/Accordion/AccordionPanel.tsx [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
 
-export const AccordionPanel: FC<AccordionPanelProps> = ({ children, flush, open }): JSX.Element => {
-  const [isOpen, setIsOpen] = useState(open);
-  const items = useMemo(
-    () => Children.map(children as ReactElement<ComponentProps<'div' | 'button'>>[], (child) => cloneElement(child)),
-    [children],
-  );
+export const AccordionPanel: FC<AccordionPanelProps> = ({ children, ...props }): JSX.Element => {
+  const { alwaysOpen } = props;
+  const [open, setOpen] = useState(props.open);
```
Comment (rluders) [inline src/lib/components/Accordion/AccordionPanel.tsx:12]: isOpen

File: src/lib/components/Accordion/AccordionTitle.tsx [hunk 3]
```diff
@@ -1,38 +1,37 @@
...
-  arrowIcon: ArrowIcon = HiChevronDown,
-  as: Heading = 'h2',
-  children,
-  ...props
-}): JSX.Element => {
+export const AccordionTitle: FC<AccordionTitleProps> = ({ as: Heading = 'h2', children, ...props }): JSX.Element => {
   const theirProps = excludeClassName(props);
 
-  const { flush, isOpen, setIsOpen } = useAccordionContext();
+  const { arrowIcon: ArrowIcon, flush, open, setOpen } = useAccordionContext();
```
Comment (rluders) [inline src/lib/components/Accordion/AccordionTitle.tsx:15]: isOpen

**Review by rluders — CHANGES_REQUESTED** [review]
About boolean values, read:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

**Comment by tulup-conner** [comment]
> About boolean values, read: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

Sure, I do that in my own projects, but pretty much every boolean in the Flowbite API doesn't start with is :)

I'm happy to change to that convention universally.

**Comment by tulup-conner** [comment]
Let me know if you want me to make the change still. The API for Accordions currently accepts an attribute `open`, not `isOpen`, which is my motivation.

**Comment by rluders** [comment]
I think that we can keep the API, but change the state. Can't we? Mostly, 'cause I think that `API != state`.

**Comment by tulup-conner** [comment]
@rluders Absolutely, I'll change these and I'll make sure to use is for variable booleans

**Review by rluders — APPROVED** [review]

**Pull request merged**